### ISSUE TITLE
Investigate if Solidity 0.5.0 will break solidity-coverage

### ISSUE THREAD

**Issue by cgewecke** [sc-forks/solidity-coverage#258]
One breaking change is the [removal of the constant key from the ABI](https://github.com/ethereum/solidity/pull/4090). However it looks like SC will be ok with this, [here](https://github.com/sc-forks/solidity-coverage/blob/master/lib/preprocessor.js#L72)? We don't really touch the ABI directly.

**Comment by NoahHydro** [comment]
One breaking change I've found is that in interfaces, marking `bytes`, `string`, etc. variables with the required `calldata` location throws an error. For example, the below throws `Expected ")", ",", comment, end of line, or whitespace but "v" found.`:

```
pragma solidity ^0.5.0;

interface MyInterface {
    function myFunction(bytes calldata myBytes);
}
```

**Comment by cgewecke** [comment]
Thanks @NoahHydro.

**Comment by elenadimitrova** [comment]
Breaking on the new `address payable` syntax as well https://circleci.com/gh/JoinColony/colonyNetwork/5422

**Comment by angus-hamill** [comment]
https://github.com/sc-forks/solidity-parser/pull/17
Opened a PR for the calldata issue here, runs successfully on my project

**Comment by cgewecke** [comment]
Yes. It did. Fixed in `0.6.0`
